spec/controllers/hq/partners_controller_spec.rb

```diff
@@ -20,26 +20,52 @@ def foobar_action
 end
 
 RSpec.describe Hq::PartnersController, type: :controller do
-  before :each do
-    sign_in instance_double(AdminUser)
-    @partner = build_stubbed :partner
-    expect(Partner).to receive(:find).and_return(@partner)
-  end
 
-  specify 'editing renders the edit view' do
-    get :edit, id: @partner.id
-    expect(response).to render_template 'edit'
-  end
+  context '#new and #create' do
+    
+    before :each do
+      sign_in instance_double(AdminUser)
+      @partner = FactoryGirl.build(:partner)
+    end
+    
+    specify 'successful create redirects to the show view' do
+      post :create, partner: @partner.attributes
+      @partner = Partner.find_by_name(@partner[:name])
+      expect(response).to redirect_to hq_partner_path(@partner)
+    end
```

Comment:
It results in this error:

```
  1) Hq::PartnersController#new and #create successful create redirects to the show view
     Failure/Error: expect(response).to redirect_to hq_partner_path(1)
       Expected response to be a <redirect>, but was <200>
```

At first I thought it means that `@partner` is failing validation, so I added a couple of parameters:

```
@partner = instance_double Partner, :id => 1, :province_id => 1, :start_date => Date.current, :save => true
```

But I'm still getting the same error.